src/settings/toml/migrations.rs

```diff
@@ -0,0 +1,499 @@
+use clap::{App, Arg, ArgGroup, ArgMatches};
+use serde::{Deserialize, Serialize};
+
+use crate::http;
+use crate::settings::global_user::GlobalUser;
+
+#[derive(Clone, Debug, Default, Deserialize, PartialEq, Serialize)]
+pub struct Migrations {
+    pub migrations: Vec<MigrationConfig>,
+}
+
+impl Migrations {
+    pub fn api_migration(
+        &self,
+        script_name: &str,
+        account_id: &str,
+        user: &GlobalUser,
+    ) -> Result<ApiMigration, failure::Error> {
+        // TODO: make api call to get most recent tag, and coalesce tags afterwards.
+        // For now, migrations will only ever have a single adhoc migration in it.
+        match &self.migrations {
+            migrations if migrations.len() == 1 => Ok(ApiMigration {
+                old_tag: None,
+                new_tag: None,
+                migration: migrations[0].migration.clone(),
+            }),
+            migrations => {
+                let list_scripts_addr = format!(
+                    "https://api.cloudflare.com/client/v4/accounts/{}/workers/scripts",
+                    account_id,
+                );
+
+                let client = http::legacy_auth_client(user);
+                let res = client.get(&list_scripts_addr).send()?;
+                let scripts = {
+                    let res: ListScriptsV4ApiResponse = res.json()?;
+                    res.result
+                };
```

Comment:
purely up to you, but I think starting a new scope here is kind of weird, expected to see
```
let res = ...
let res_json: ... = res.json()?;
let scripts = res.result;
```
but it really doesn't matter

Reply:
I quite like using scopes to avoid having to name intermediate results

Reply:
correct me if I'm mistaken or don't have the full context, but can't you avoid the intermediate result and still try the result like this? 

```rust
let scripts = res.json::<ListScriptsV4ApiResponse>()?.result;
```

(not tested !)